src/super_gradients/training/datasets/segmentation_datasets/mapillary_dataset.py

```diff
@@ -0,0 +1,115 @@
+            ├── config_v2.0.json
+            ├── training
+                ├── images
+                    ├── {image_name}.jpg            # RGB images
+                ├── v1.2
+                    ├── labels
+                        ├── {image_name}.jpg        # Target masks
+                ├── v2.0
+                    ├── labels
+                        ├── {image_name}.jpg        # Target masks
+            ├── validation
+            ├── testing
+        Note that there are two versions currently available for this dataset, `v1.2` and `v2.0`, the difference according
+        to the change log is as follows:
+            - Expanded the set of labels to 124 classes (70 instance-specific, 46 stuff, 8 void or crowd).
+            - Added raw polygonal annotations as json files. These reflect the ordering in which the segments where
+                annotated by the original annotators, i.e. approximately from the background towards the camera.
+        The common practice is to use the 65 categorical labels from v1.2 and older.
+    """
+
+    IGNORE_LABEL_V1_2 = 65
+    IGNORE_LABEL_V2_0 = 123
+
+    def __init__(
+        self,
+        root_dir: str,
+        config_file: str,
+        samples_sub_directory: str,
+        targets_sub_directory: str,
+        sample_extension: str = ".jpg",
+        target_extension: str = ".png",
+        **kwargs,
+    ):
+        self.samples_sub_directory = samples_sub_directory
+        self.targets_sub_directory = targets_sub_directory
+        self.target_extension = target_extension
+        self.sample_extension = sample_extension
+        # FIXME - Must pass list_file, due to double inheritance error when using DirectoryDataset. See the bug report
+        super().__init__(
+            root=root_dir,
+            samples_sub_directory=samples_sub_directory,
+            targets_sub_directory=targets_sub_directory,
+            list_file="",
+            target_extension=target_extension,
+            **kwargs,
+        )
+
+        # read in config file
+        with open(os.path.join(self.root, config_file), "r") as f:
+            config = json.load(f)
+        self.labels = config["labels"]
+        self.label_colors = [label["color"] for label in self.labels]
+        self.label_names = [label["readable"].replace(" ", "_") for label in self.labels]
+        # Ignore labels is called `Unlabeled` in config files
+        self.ignore_label = self.label_names.index("Unlabeled")
+        # SG format requires returning classes as label names without ignore labels, it is also often used to calculate
+        # the num of classes.
+        self.classes = self.label_names[:-1]
+
+    def _generate_samples_and_targets(self):
+        samples_dir = os.path.join(self.root, self.samples_sub_directory)
+        labels_dir = os.path.join(self.root, self.targets_sub_directory)
+
+        sample_names = [n for n in os.listdir(samples_dir) if n.endswith(self.sample_extension)]
```

Comment:
`os.listdir` does not guarantee that order of files would be the same in different OS. 
`sorted(os.listdir)` would solve this problem.

Reply:
This shouldn't matter as long as results are consistent between two calls.
Anyway, your repo your rules - [d1b37d2](https://github.com/Deci-AI/super-gradients/pull/594/commits/d1b37d2ff00525655b0de31b1eeed09af4724a2f)